extensions/aws/src/main/java/io/opentelemetry/extension/aws/AwsXrayPropagator.java

```diff
@@ -210,16 +211,20 @@ private static String parseTraceId(String xrayTraceId) {
     }
 
     // Check delimiters
-    if (xrayTraceId.charAt(TRACE_ID_DELIMITER_INDEX_1) != TRACE_ID_DELIMITER
-        || xrayTraceId.charAt(TRACE_ID_DELIMITER_INDEX_2) != TRACE_ID_DELIMITER) {
+    int firstDelimiter = xrayTraceId.indexOf(TRACE_ID_DELIMITER);
```

Comment:
Can we keep the original code and add a parseSlow for the short case? It's against spec so not fair all parsing gets a bit slower :)

Reply:
Definitely! Please have a look at the updated version.